api/v1alpha1/virtualmachineimage_conversion.go

```diff
@@ -448,6 +458,14 @@ func (src *ClusterVirtualMachineImage) ConvertTo(dstRaw ctrlconversion.Hub) erro
 		return err
 	}
 
+	// Manually restore data.
+	restored := &vmopv1.ClusterVirtualMachineImage{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
+	}
+
+	dst.Status.DiskInfo = restored.Status.DiskInfo
```

Comment:
Why not just?

```golang
dst.Status = restored.Status
```

Reply:
Wouldn't that overwrite what was done in `convert_v1alpha1_VirtualMachineImageSpec_To_v1alpha3_VirtualMachineImageStatus` on line 452?